unison-src/transcripts/destructuring-binds.md

```diff
@@ -0,0 +1,72 @@
+# Destructuring binds
+
+```ucm:hide
+.> builtins.merge
+```
+
+Here's a couple examples:
+
+```unison
+ex1 : (a,b,(Nat,Nat)) -> Nat
+ex1 tup =
+  (a, b, (c,d)) = tup
+  c + d
+```
+
+```ucm
+.> add
```

Comment:
It looks like if the expression could be rendered as `cases` or destructuring bind, it chooses `cases`. I'm not sure what I think about that. Maybe it's fine?

Reply:
Yeah, the `cases` version looks OK to me, maybe better since it doesn't repeat `tup` as in
```
ex1 : (a, b, (Nat, Nat)) -> Nat
ex1 tup =
  (a, b, (c,d)) = tup
  c + d
```
Also it would be tricky to change because the `cases` stuff is done early in `specialCases` which is already a bit hairy.

Reply:
Probably worth making the transcript use a different example though.

Reply:
Yeah... I'm going to leave it in there to mention the behavior, but am adding another example that doesn't overlap with `cases`.